### PR TITLE
Ci update

### PR DESCRIPTION
**Pull request by gquintard**
move things around a bit, but most importantly removes `centos:8` in favor of `fedora:latest`

### CODE REVIEW

**Comment by gquintard** [comment]
it will need https://github.com/varnishcache/pkg-varnish-cache/pull/153 (or something like it) for the packages to build though

File: .circleci/config.yml [hunk 1]
```diff
@@ -176,14 +176,15 @@ jobs:
...
-            docker build -t docker_img .
-            docker run --rm -it -e PARAM_DIST=$(echo "<< parameters.platform >>" | cut -d: -f1) -e PARAM_RELEASE=$(echo "<< parameters.platform >>" | cut -d: -f2) -v$(pwd):/varnish-cache docker_img
+            docker run \
+              --rm \
+              -it \
+              -e PARAM_DIST=$(echo "<< parameters.platform >>" | cut -d: -f1) \
+              -e PARAM_RELEASE=$(echo "<< parameters.platform >>" | cut -d: -f2) \
+              -v$(pwd):/varnish-cache \
+              $IMG \
+              /varnish-cache/make-$EXT-packages.sh
```
Comment (Dridi) [inline .circleci/config.yml:187]: Can't we skip the copy (technically tainting the local git clone) and run this instead?

```
[...] \
    $IMG \
    /varnish-cache/.circleci/make-$EXT-packages.sh
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -353,9 +351,13 @@ workflows:
...
-          name: distcheck_centos_8
-          dist: centos
-          release: "8"
+          name: distcheck_fedora_stable
+          dist: fedora
+          release: latest
+      - distcheck:
+          name: distcheck_fedora_rawhide
+          dist: fedora
+          release: rawhide
```
Comment (Dridi) [inline .circleci/config.yml:356]: I would prefer `distcheck_fedora_latest` to match the image tag.

File: .circleci/make-rpm-packages.sh [hunk 3]
```diff
@@ -13,13 +13,9 @@ elif [ -z "$PARAM_DIST" ]; then
...
 
-yum install -y epel-release
 
 if [ "$PARAM_DIST" = centos ]; then
-  if [ "$PARAM_RELEASE" = 8 ]; then
-      dnf install -y 'dnf-command(config-manager)'
-      yum config-manager --set-enabled powertools
-  fi
+    yum install -y epel-release
 fi
```
Comment (Dridi) [inline .circleci/make-rpm-packages.sh:19]: This adjustment should happen before fedora's introduction, modulus el8 removal.

**Review by Dridi — CHANGES_REQUESTED** [review]
I pushed the arch changes and the machine executor update, but please leave the centos:8 removal out for now.

File: .circleci/config.yml (hunk 2 above)
Comment (gquintard) [reply .circleci/config.yml:356]: That would be smarter indeed

**Comment by gquintard** [comment]
Are you sure you want to keep CentOS:8? The repos seem to have been failing for some time now

**Comment by Dridi** [comment]
> Are you sure you want to keep CentOS:8? The repos seem to have been failing for some time now

I want to handle centos-8 separately, and I'm considering centos-stream (and/or centos-stream[89]) as a replacement.

**Comment by gquintard** [comment]
tweaked and force-pushed. I left this change https://github.com/varnishcache/varnish-cache/pull/3780/commits/e39438ba919cdda73b7b738f5a93dd2d92717f89#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47L223 because it works now, and I don't feel like shooting in the dark trying to get compatibility with a distribution I can't test

File: .circleci/config.yml (hunk 1 above)
Comment (Dridi) [reply .circleci/config.yml]: Why mark this one as resolved?

File: .circleci/config.yml [hunk 4]
```diff
@@ -176,15 +176,14 @@ jobs:
...
-            cp .circleci/make-$EXT-packages.sh .
             docker run \
               --rm \
               -it \
               -e PARAM_DIST=$(echo "<< parameters.platform >>" | cut -d: -f1) \
               -e PARAM_RELEASE=$(echo "<< parameters.platform >>" | cut -d: -f2) \
               -v$(pwd):/varnish-cache \
               $IMG \
-              /varnish-cache/make-$EXT-packages.sh
+              /varnish-cache/.circleci/make-$EXT-packages.sh
```
Comment (Dridi) [inline .circleci/config.yml:186]: I see, you did that in a followup commit instead.

**Pull request closed without merging**